src/platform/w3dengine/client/w3dbridgebuffer.cpp

```diff
@@ -0,0 +1,790 @@
+                        lights);
+
+                    if (!vcount) {
+                        captainslog_dbgassert(vcount, "W3DBridge::GetIndicesNVertices(). Vertex overflow.");
+                        return;
+                    }
+
+                    icount = Get_Model_Indices(destination_ib, *cur_index, *cur_vertex, m_sectionMesh);
+
+                    if (!icount) {
+                        captainslog_dbgassert(icount, "W3DBridge::GetIndicesNVertices(). Index overflow.");
+                        return;
+                    }
+
+                    *cur_index += icount;
+                    *cur_vertex += vcount;
+                    m_numVertex += vcount;
+                    m_numPolygons += icount / 3;
+                }
+
+                vcount = Get_Model_Vertices(destination_vb,
+                    *cur_vertex,
+                    (float)sections * f2 + x_offset,
+                    vec,
+                    vec_normal,
+                    vec_z,
+                    m_start,
+                    m_rightMtx,
+                    m_rightMesh,
+                    lights);
+
+                if (!vcount) {
+                    captainslog_dbgassert(vcount, "W3DBridge::GetIndicesNVertices(). Vertex overflow.");
+                } else {
+                    icount = Get_Model_Indices(destination_ib, *cur_index, *cur_vertex, m_rightMesh);
+
+                    if (!icount) {
+                        captainslog_dbgassert(icount, "W3DBridge::GetIndicesNVertices(). Index overflow.");
+                    } else {
+                        *cur_index += icount;
+                        *cur_vertex += vcount;
+                        m_numVertex += vcount;
+                        m_numPolygons += icount / 3;
+                    }
+                }
+            }
+        }
+    } else {
+        int vcount = Get_Model_Vertices_Fixed(destination_vb, *cur_vertex, m_leftMtx, m_leftMesh, lights);
+
+        if (!vcount) {
+            captainslog_dbgassert(vcount, "W3DBridge::GetIndicesNVertices(). Vertex overflow.");
+        } else {
+            int icount = Get_Model_Indices(destination_ib, *cur_index, *cur_vertex, m_leftMesh);
+
+            if (!icount) {
+                captainslog_dbgassert(icount, "W3DBridge::GetIndicesNVertices(). Index overflow.");
+            } else {
+                *cur_index += icount;
+                *cur_vertex += vcount;
+                m_numVertex += vcount;
+                m_numPolygons += icount / 3;
+            }
+        }
+    }
+}
+
+void W3DBridgeBuffer::Cull(CameraClass *camera)
+{
+    m_anythingChanged = m_updateVis;
+
+    for (int i = 0; i < m_numBridges; i++) {
+        if (m_bridges[i].Cull_Bridge(camera)) {
+            m_anythingChanged = true;
+        }
+    }
+}
+
+void W3DBridgeBuffer::Load_Bridges_In_Vertex_And_Index_Buffers(RefMultiListIterator<class RenderObjClass> *lights)
+{
+#ifdef BUILD_WITH_D3D8
+    if (m_indexBridge) {

```

Comment:
Not a fan of pyramids personally. Could you just return early and make this not a pyramid.